src/main/kotlin/com/autonomousapps/subplugin/ProjectPlugin.kt

```diff
@@ -509,6 +516,20 @@ internal class ProjectPlugin(private val project: Project) {
     }
   }
 
+  private fun Project.executeOnAllowedVariants(variant: String, action: () -> Unit) {
+    val allowedVariants = getAllowedVariants()
+    val ignoredVariants = getIgnoredVariants()
+    val commonVariants = allowedVariants.intersect(ignoredVariants)
+
+    if (commonVariants.isNotEmpty()) {
+      throw GradleException("Allowed and ignored variants must be different. Issue found in variant(s): $commonVariants")
+    } else if ((allowedVariants.isEmpty() && ignoredVariants.isEmpty()) ||
+          (!ignoredVariants.contains(variant) && allowedVariants.contains(variant))) {
```

Comment:
I don't think this quite works. If allowed variants is empty, then the variant should be permitted by default (unless explicitly ignored). Similarly if ignored variants is empty, then the variant should be permitted by default (unless allowed variants is not empty).

Alternatively, I think I'm not convinced that we want _both_ an allow-list and an ignore-list. Is there a reason for that complexity? Why not just pick one?

Reply:
will go through again with the condition
the reason of having both allowed and ignored list - in case of a big project, one can have many variants like beta, gms, hms, internal, prod... so if one don't want to execute the task on prod variants then instead of adding all the possible variants that needs to be allowed, one only has to add the prod variant in the ignored list